### DISCUSSION TITLE
[HttpFoundation] About sorting query parameters in Request::getUri()

### DISCUSSION THREAD

**Discussion by phcorp** [symfony/symfony#50593] [Q&A] [answered]
Hello,
I recently discovered that query parameters position was not conserved when accessing it via `Request::getUri()` and I wonder why that is the case. ~~This change was introduced since 4.1 by [this commit](https://github.com/symfony/http-foundation/commit/aa64c6c805c45a824ae9a66c42e3267e2e2e621a).~~ (EDIT: the change is prior to this commit, see the following comment) AFAIK this method should provide the current page URI and I don't quite understand why the query parameters order is different in the URI provided by this method from the request URI. If the URI from `Request::getUri()` was reused for example to generate a canonical URL, this may cause caching issues.
I'd really like to have more information about it, I'm probably missing the context.

**Comment by ghost** [comment]
Actually the query parameters position was not conserved both before and after this commit, see https://3v4l.org/ukWam.
I agree that it should not be reordered. There's no reason to.

**Comment by phcorp** [comment, marked as answer]
So here are the conclusions I came to:
- there will be no problem with a canonical URL which will always be the same with the sorted parameters ;
- if there is a cache proxy between my users' browser and the PHP server (e.g. varnish), then some logic needs to be implemented to avoid caching the same URL twice with parameters in a different order ; this may be the responsibility of the dns or proxy server.

It seems to me that this is more of a problem that needs to be dealt with upstream outside symfony.
I'm closing the discussion
